message TestRef {
	optional uint32 field1 = 1;
	optional int64 field2 = 2;
	optional double field3 = 3;
	optional string field4 = 4;
}
